client/utils/loadComponent.js

```diff
@@ -0,0 +1,30 @@
+import React, { Component } from 'react';
+
+const loadComponent = (importCall) => {
+  return class extends Component {
+    constructor(props) {
+      super(props);
+      this.state = {
+        component: () => <div>Loading...</div>
+      };
+    }
+
+    handleLoadComponent = async () => {
+      const importedComponent = await importCall
```

Comment:
Add a semicolon here.

Reply:
Also why do we need to do this await here?

Reply:
@BolajiOlajide It's to get the component (resolved promise) better instead of using `.then() promise` that takes more line of code.

Reply:
Haha! Make sense.

As you were then.